### ISSUE TITLE
Grid layout when combining * and Auto definitions

### ISSUE THREAD

**Issue by arifu-dev** [taublast/DrawnUi.Maui#110]
This was tested on the latest 1.2.4.7 version.

This is a rather specific scenario: when there is a mix in the definition of `*` and `Auto`, and the `Auto` row contains a layout without a defined size then `*` will not take up any space and `Auto` will simply fill the remaining space.
Here is an example:
```
<draw:Canvas>
    <draw:SkiaLayout Type="Grid" RowDefinitions="*, 70, Auto" BackgroundColor="Green">
        <draw:SkiaShape Type="Rectangle" BackgroundColor="Red"/>
        <draw:SkiaShape Type="Rectangle" BackgroundColor="DarkCyan" Grid.Row="1"/>
        <draw:SkiaShape Type="Rectangle" BackgroundColor="Blue" Grid.Row="2">
            <draw:SkiaLayout Type="Grid" ColumnDefinitions="*, *" HorizontalOptions="Fill">
                <draw:SkiaLabel Text="test" HorizontalOptions="Center"/>
                <draw:SkiaLabel Text="test" HorizontalOptions="Center" Grid.Column="1"/>
            </draw:SkiaLayout>
        </draw:SkiaShape>
    </draw:SkiaLayout>
</draw:Canvas>
```

Which makes the `Auto` row take up all the remaining space:

![image](https://github.com/user-attachments/assets/c21db922-3b90-4406-b8ca-901d9949a1e8)

In MAUI `Auto` is supposed to take up enough space to fit it's children, and `*` is supposed to take up the remaining space. This is how it works here as well if either the SkiaShape in row=2 or it's child SkiaLayout has a `HeightRequest` specified.

Here's a MAUI example for reference:
```
<Grid RowDefinitions="*, 70, Auto" BackgroundColor="Green">
    <Frame BackgroundColor="Red"/>
    <Frame BackgroundColor="DarkCyan" Grid.Row="1"/>
    <Frame BackgroundColor="Blue" Grid.Row="2">
        <Grid ColumnDefinitions="*, *" HorizontalOptions="Fill">
        <Label Text="test" HorizontalOptions="Center"/>
        <Label Text="test" HorizontalOptions="Center" Grid.Column="1"/>
        </Grid>
    </Frame>
</Grid>
```

![image](https://github.com/user-attachments/assets/b7be2644-bb56-48d9-b753-d408d45505fd)

**Comment by taublast** [comment]
Are you sure this is not coming from Maui having controls h/v options fill and drawnui having them as start? What would happen if you set horizontal and vertical options to fill to children controls?

Normally in drawnui the following will take zero space in any layout:

<draw:SkiaShape Type="Rectangle" BackgroundColor="Red"/>

**Comment by arifu-dev** [comment]
Yep, I'm sure.
Setting both horizontal and vertical options to fill for the red SkiaShape changes nothing.
Meanwhile in the MAUI example I can remove the red frame completely it will still reserve all the remaining space for the row, just with nothing in it (green is the background for the top grid)

![image](https://github.com/user-attachments/assets/c74700da-e13a-457b-862d-9b0114dc70a2)

**Comment by arifu-dev** [comment]
And it also works correctly if I set a SkiaLabel as the SkiaShape's content:

```
<draw:Canvas>
    <draw:SkiaLayout Type="Grid" RowDefinitions="*, 70, Auto" BackgroundColor="Green">
        <draw:SkiaShape Type="Rectangle" BackgroundColor="Red"/>
        <draw:SkiaShape Type="Rectangle" BackgroundColor="DarkCyan" Grid.Row="1"/>
        <draw:SkiaShape Type="Rectangle" BackgroundColor="Blue" Grid.Row="2">
            <draw:SkiaLabel Text="Test"/>
        </draw:SkiaShape>
    </draw:SkiaLayout>
</draw:Canvas>
```

![image](https://github.com/user-attachments/assets/6a3366d3-c96c-4144-bdfc-22a477705a50)

The issue appears only if the SkiaShape's content is a layout.

**Comment by taublast** [comment]
Hah, so in the end the BUG is that "Grid auto-height (to be checked for width) is never of content size but is of full available size".
![image](https://github.com/user-attachments/assets/e4f10de7-f059-4c52-af78-0b6e2686ae84)
As we see Grid just took all the available height for some reason, instead of text size. Going into "hunt bug and destroy" mode :)

**Comment by taublast** [comment]
And then: my default GridLength for a row is Star. So by defaults it takes all available space. So it's like "by design". I think we would need to change defaults for a col/row to Auto that might do it.

**Comment by taublast** [comment]
So i'll set defaults to Auto, that makes bugs in your examples disappear:

![image](https://github.com/user-attachments/assets/cee0d587-cb98-4480-af85-69edabdf4ab2)


Also we have some props that MAUI doesn't have: `DefaultColumnDefinition` and `DefaultRowDefinition` that one could use if they don't like defaults. They were introduced for cases like, for example if you want all rows to be like 60 height, you set this to 60 and then just spam adding new rows and that all will be of 60 automatically without the need to add specific rowdefinitions. :)

You can simulate this fix by setting manually `DefaultRowDefinition="Auto"` in your grids today, until I push this in new nuget.

**Comment by taublast** [comment]
A lill offtopic, in the upcoming nuget 1.2.4.8 containing this fix:

* Grid Auto definitions fix (defaults are now Auto). Use `DefaultRowDefinition` and `DefaultColumnDefinition` if needed.

 we'll also have a new feature: 

* Implemented `Background` property inside `SkiaControl`. You can use MAUI usual syntax for gradients and solid color backgrounds.

![image](https://github.com/user-attachments/assets/08b9f3ca-27e3-4219-bb83-e916d66ff021)

**Comment by taublast** [comment]
Fixed with 55226dbbfe33af02864e56edb3003785c595de51

**Comment by arifu-dev** [comment]
Hey thanks so much for the quick fix, indeed changing the default definitions fixes the problem!
